services/core/shared/core/compat/sdk_v5/network.js

```diff
@@ -15,26 +15,53 @@
  */
 const coreApi = require('./coreApi');
 const { resolvemoduleAssets } = require('../common/constants');
+const { parseToJSONCompatObj } = require('../../../jsonTools');
 
 const requestApi = coreApi.requestRetry;
 
+// As in https://github.com/LiskHQ/lisk-sdk/blob/v5.1.4/elements/lisk-chain/src/block_reward.ts
+const calculateMilestone = (height, blockRewardArgs) => {
+	const distance = Math.floor(blockRewardArgs.distance);
+	const location = Math.trunc((height - blockRewardArgs.offset) / distance);
+	const lastMile = blockRewardArgs.milestones[blockRewardArgs.milestones.length - 1];
+
+	if (location > blockRewardArgs.milestones.length - 1) {
+		return blockRewardArgs.milestones.lastIndexOf(lastMile);
+	}
+
+	return location;
+};
+
+const calculateDefaultReward = (height, blockRewardArgs) => {
+	if (height < blockRewardArgs.offset) {
+		return BigInt(0);
+	}
+
+	return blockRewardArgs.milestones[calculateMilestone(height, blockRewardArgs)];
+};
```

Comment:
This can be simplified like this (Optional):
```suggestion
const calculateDefaultReward = (height, blockRewardArgs) => (height < blockRewardArgs.offset)
	? BigInt(0)
	: blockRewardArgs.milestones[calculateMilestone(height, blockRewardArgs)];
```

Reply:
I'm not modifying them as these methods have been taken from the Lisk SDK repo.